dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListenerManager.cpp

```diff
@@ -37,15 +38,22 @@ void NotificationListenerManager::RemoveListener(std::wstring appId)
 {
     THROW_HR_IF(E_INVALIDARG, appId.empty());
 
-    auto lock = m_lock.lock_exclusive();
+    auto lock{ m_lock.lock_exclusive() };
 
     LOG_IF_FAILED(PushNotifications_UnregisterNotificationSinkForFullTrustApplication(appId.c_str()));
 
     m_notificationListeners.erase(appId);
 }
 
+bool NotificationListenerManager::HasSinkForAppId(std::wstring const& appId)
+{
+    auto lock{ m_lock.lock_shared() };
```

Comment:
I'm curious about this pattern. You are calling this in an exclusive lock while the method has a shared lock. You are infact deadlocking.

Reply:
Actually never mind. This is a public helper. I think you are good